### ISSUE TITLE
Explain how `on_schema_change` handles new or removed columns for an incremental model

### ISSUE THREAD

**Issue by dbeatty10** [fishtown-analytics/docs.getdbt.com#2902]
### Contributions

- [X] I have read the contribution docs, and understand what's expected of me.

### Link to the page on docs.getdbt.com requiring updates

https://docs.getdbt.com/docs/build/incremental-models#what-if-the-columns-of-my-incremental-model-change

### What part(s) of the page would you like to see updated?

@Fleid noted [here](https://github.com/dbt-labs/dbt-adapters/issues/601) that the following is a more complete and accurate description of how the value of `on_schema_change` behaves for incremental models when:
1. new columns are added, and
2. when columns are removed

|`on_schema_change`|In case of new columns...|In case of removed columns...|
|-|-|-|
|`ignore`|columns ignored|run failed|
|`fail`|run failed|run failed|
|`append_new_columns`|columns added|columns ignored|
|`sync_all_columns`|columns added|columns removed|

**Comment by awong-evoiq** [comment]
This is incorrect. The docs already state that `ignore` should not fail when the incremental table schema has columns that the existing table (in the database) does not. https://docs.getdbt.com/docs/build/incremental-models#default-behavior.

The problem in the mentioned issue is that dbt fails when the incremental update has a new column. It behaves the same as the `fail` option. If you look at @Fleid's Feb 13 response to that issue, he identifies the root cause of why the `ignore` setting fails. He recommends fixing it, and renaming the current behavior as `classic` (although I don't see how `classic` is different from `fail`).

**Comment by mirnawong1** [comment]
> [!NOTE]
> 🤖 **AI Triage Suggestion** — Automated analysis. A human reviewer will make the final call on labels and priority.

**Summary:** The docs don't clearly explain what happens to your table columns when you add or remove them from an incremental model. This issue asks us to add a simple table showing what each `on_schema_change` setting does in both cases — and the table content is already written and ready to add.

| | Suggestion | Reasoning |
|---|---|---|
| **Priority** | HIGH | Over 3 years old; the fix is well-scoped with the content already drafted — leaving it unmerged means users continue encountering undocumented behavior on a commonly-used config. |
| **Size** | XS | Single page update — insert a pre-written table with minimal surrounding edits. |
| **Product area** | Core | Directly about incremental model behavior, a Core resource type. |

_Labels won't be applied automatically. Apply labels directly to confirm or adjust._
